src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpClientJsonExtensions.Get.AsyncEnumerable.cs

```diff
@@ -135,12 +135,29 @@ public static partial class HttpClientJsonExtensions
             JsonSerializerOptions? options,
             CancellationToken cancellationToken)
         {
-            options ??= JsonSerializerOptions.Default;
-            options.MakeReadOnly();
+            if (client is null)
```

Comment:
Private helpers should not be doing null validation. I would recommend replacing this will a `Debug.Assert(client != null);` statement and make the actual checks in the public methods that call into the helpers. This also avoids redundant checking that `client` is null in cases where the IAsyncEnumerable will be iterated over multiple times.

Reply:
updated